src/editor/hotkeys.cpp

```diff
@@ -25,7 +81,43 @@ void CodeEditor::keyPressEvent(QKeyEvent *e)
            this->verticalScrollBar();
            return;
         }
+        if(e->key() == Qt::Key_Slash){
+           //QPlainTextEdit::keyPressEvent(e);
+            pressSlesh=true;
+        }
+        if(e->key() == Qt::Key_Asterisk && pressSlesh){
+            QPlainTextEdit::keyPressEvent(e);
+            this->insertPlainText("*/");
+            this->moveCursor(QTextCursor::Left);
+            this->moveCursor(QTextCursor::Left);
+            this->verticalScrollBar();
+            return;
 
+        }
+        if(e->key() == Qt::Key_ParenLeft)
+        {
+            QPlainTextEdit::keyPressEvent(e);
+            this->insertPlainText(")");
+            this->moveCursor(QTextCursor::Left);
+            this->verticalScrollBar();
+            return;
+        }
+        if(e->key() == Qt::Key_Apostrophe)
+        {
+            QPlainTextEdit::keyPressEvent(e);
+            this->insertPlainText("\'");
+            this->moveCursor(QTextCursor::Left);
+            this->verticalScrollBar();
+            return;
+        }
+        if(e->key() == Qt::Key_QuoteDbl)
+        {
+            QPlainTextEdit::keyPressEvent(e);
+            this->insertPlainText("\"");
+            this->moveCursor(QTextCursor::Left);
+            this->verticalScrollBar();
+            return;
+        }
         if((e->key() == Qt::Key_Plus && e->modifiers() & Qt::ControlModifier)//ctrl & +
                 && currentZoom <= 150)// forbid to zoom so much
         {
```

Comment:
Code in the if the statement is almost the same. I think you can move some part of code outside of if statement and avoid code duplication.